core/src/main/java/com/oath/oak/ZeroCopyMap.java

```diff
@@ -0,0 +1,110 @@
+     * @throws IllegalArgumentException if the specified key is out of bounds
+     */
+    OakRBuffer get(K key);
+
+    /**
+     * Removes the mapping for a key from this map if it is present.
+     *
+     * @param key key whose mapping is to be removed from the map
+     * @throws NullPointerException     if the specified key is null
+     * @throws IllegalArgumentException if the specified key is out of bounds
+     */
+    void remove(Object key);
+
+    /**
+     * If the specified key is not already associated
+     * with a value, associate it with the given value.
+     * Creates a copy of the value in the map.
+     *
+     * @param key   key with which the specified value is to be associated
+     * @param value value to be associated with the specified key
+     * @return {@code true} if there was no mapping for the key
+     * @throws NullPointerException     if the specified key or value is null
+     * @throws IllegalArgumentException if the specified key is out of bounds
+     */
+    boolean putIfAbsent(K key, V value);
+
+    /**
+     * Updates the value for the specified key
+     *
+     * @param key      key with which the calculation is to be associated
+     * @param computer for computing the new value
+     * @return {@code false} if there was no mapping for the key
+     * @throws NullPointerException     if the specified key or the function is null
+     * @throws IllegalArgumentException if the specified key is out of bounds
+     */
+    boolean computeIfPresent(K key, Consumer<OakWBuffer> computer);
+
+
+    /**
+     * If the specified key is not already associated
+     * with a value, associate it with a constructed value.
+     * Else, updates the value for the specified key.
+     *
+     * @param key      key with which the specified value is to be associated
+     * @param value    value to be associated with the specified key
+     * @param computer for computing the new value when the key is present
+     * @throws NullPointerException     if any of the parameters is null
+     * @throws IllegalArgumentException if the specified key is out of bounds
+     */
+    void putIfAbsentComputeIfPresent(K key, V value, Consumer<OakWBuffer> computer);
+
+    /**
+     * Returns a {@link Set} view of read only ByteBuffers containing the
+     * serialized keys stored in this map.
+     *
+     * @return a set view of the serialized keys contained in this map
+     */
+    Set<OakRBuffer> keySet();
+
+    /**
+     * Returns a {@link Collection} view of read only buffers containing the
+     * serialized values stored in this map.
+     *
+     * @return a collection view of the values contained in this map
+     */
+    Collection<OakRBuffer> values();
+
+    /**
+     * Returns a {@link Set} view of the serialized mappings contained in this
+     * map.
+     *
+     * @return a set view of the serialized mappings contained in this map
+     */
+    Set<Map.Entry<OakRBuffer, OakRBuffer>> entrySet();
```

Comment:
In druid i need to have an iterator on entries that has the values as ByteBuffers.
How is this possible in the new API? 
If we dont want to change anything the only way to do this is to add a "getByteBuffer()" function in OakRBuffer